pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -0,0 +1,113 @@
+package persistentvolume
+
+import (
+	"strings"
+
+	"github.com/openshift/ocs-operator/pkg/controller/util"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	csiExpansionSecretName      = "csi.storage.k8s.io/controller-expand-secret-name"
+	csiExpansionSecretNamespace = "csi.storage.k8s.io/controller-expand-secret-namespace"
+)
+
+var log = logf.Log.WithName("controller_persistentvolume")
+
+// Add creates a new PersistentVolume Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcilePersistentVolume{
+		client:    mgr.GetClient(),
+		scheme:    mgr.GetScheme(),
+		reqLogger: log,
+	}
+
+	return r
+}
+
+// reconcilePV
+func reconcilePV(obj runtime.Object) bool {
+	pv, ok := obj.(*corev1.PersistentVolume)
+	if !ok {
+		return false
+	}
+
+	if pv.Spec.CSI != nil && strings.Contains(pv.Spec.CSI.Driver, "csi.ceph.com") {
```

Comment:
can this cause issue if some driver comes with csi.ceph.com in their driver name? as we have the exact driver names in ocs-operator can we do full name comparison?

Reply:
We don't have the exact driver names in this controller. The full driver names are generated from the Rook-Ceph operator's namespace. We don't have access to the Rook-Ceph operator, since we don't want to assume the name of the operator's Deployment. Still, I can make it a little more explicit by specifying `rbd` and `cephfs`. If we could have the suffixes exposed via the go code, that would be even better, but right now they're just strings in the code: https://github.com/rook/rook/blob/662bde6d4a7cdd51090bea8bb73b314376f47ad2/pkg/operator/ceph/csi/spec.go#L220-L221

Reply:
ocs-operator creates default cephfs and rbd storageclass and snapshotter for the csi drivers . Is it possible to do same thing here?

Reply:
Okay got it.

Reply:
Addressed.